Vagrantfile

```diff
@@ -57,11 +63,43 @@
 #
 # Customization Options
 # =====================
-ALAVETELI_FQDN = ENV['ALAVETELI_VAGRANT_FQDN'] || "alaveteli.10.10.10.30.xip.io"
-ALAVETELI_MEMORY = ENV['ALAVETELI_VAGRANT_MEMORY'] || 1536
-ALAVETELI_THEMES_DIR = ENV['ALAVETELI_THEMES_DIR'] || '../alaveteli-themes'
-ALAVETELI_OS = ENV['ALAVETELI_VAGRANT_OS'] || 'wheezy64'
-ALAVETELI_USE_NFS = ENV['ALAVETELI_VAGRANT_USE_NFS'] || false
+#
+# Defaults can be overridden either in `.vagrant.yml` with the same key name, or
+# via the environment by prefixing the key with `ALAVETELI_VAGRANT_` and
+# upcasing. Boolean values can be set to `false` in the environment with "0",
+# "false" or "no". The environment takes precedence over `.vagrant.yml`.
+DEFAULTS = {
+  'fqdn' => 'alaveteli.10.10.10.30.xip.io',
+  'ip' => '10.10.10.30',
+  'memory' => 1536,
+  'themes_dir' => '../alaveteli-themes',
+  'os' => 'wheezy64',
+  'use_nfs' => false,
+  'show_settings' => false
+}.freeze
+
+env = DEFAULTS.keys.reduce({}) do |memo, key|
+  value = ENV["ALAVETELI_VAGRANT_#{ key.upcase }"]
+  value = false if %w(0 false no).include?(value)
+  memo[key] = value if value
+  memo
+end
+
+settings_file_path = File.dirname(__FILE__) + '/.vagrant.yml'
+settings_file = if File.exist?(settings_file_path)
+  YAML.load(File.read(settings_file_path))
+else
+  {}
+end
+
+SETTINGS = DEFAULTS.merge(settings_file).merge(env).freeze
+
+if SETTINGS['show_settings']
+  puts 'Current machine settings:'
+  puts "\n"
+  pp SETTINGS
+  puts "\n"
+end
```

Comment:
this is pretty cool - I like being able to see what settings I'm supposed to have :cake: